services/vector_db_service.py

```
@@ -0,0 +1,42 @@
+        if index_name not in existing:
+            self.pc.create_index(
+                name=index_name,
+                dimension=dimension,
+                spec=ServerlessSpec(cloud=self.cloud, region=self.region),
+            )
+        self.index = self.pc.Index(index_name)
+
+    def upsert_vectors(self, vectors: List[List[float]], ids: List[str]):
+        # Input validation
+        if vectors is None or ids is None:
+            raise ValueError("vectors and ids must not be None.")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)